core/src/main/java/nl/nn/adapterframework/pipes/WsdlXmlValidator.java

```diff
@@ -197,8 +195,42 @@ public void configure() throws ConfigurationException {
 		super.configure();
 	}
 
+	@Override
+	protected PipeForward validate(Message messageToValidate, PipeLineSession session, boolean responseMode, String messageRoot) throws XmlValidatorException, PipeRunException, ConfigurationException {
+		String soapAction = session.get(SoapBindingConstants.SOAP_ACTION, "");
+		if(StringUtils.isNotEmpty(soapAction)) {
+			String soapBodyFromSoapAction = getSoapBodyFromSoapAction(soapAction);
+			if(StringUtils.compare(getSoapBody(), soapBodyFromSoapAction) != 0) {
```

Comment:
I would expect that the soapBodyFromSoapAction was different for request or responseMode, or is only valid for one of them.
I also think that the compare against getSoapBody is not valid. Shouldn't that be against messageRoot?